internal/login/handler/mail_verify_handler.go

```diff
@@ -0,0 +1,91 @@
+package handler
+
+import (
+	"github.com/caos/zitadel/internal/auth_request/model"
+	"net/http"
+)
+
+const (
+	queryCode   = "code"
+	queryUserID = "userID"
+
+	tmplMailVerification = "mail_verification"
+	tmplMailVerified     = "mail_verified"
+)
+
+type mailVerificationFormData struct {
+	Code   string `schema:"code"`
+	UserID string `schema:"userID"`
+	Resend bool   `schema:"resend"`
+}
+
+type mailVerificationData struct {
+	baseData
+	UserID string
+}
+
+func (l *Login) handleMailVerification(w http.ResponseWriter, r *http.Request) {
+	userID := r.FormValue(queryUserID)
+	code := r.FormValue(queryCode)
+	if code != "" {
+		l.checkMailCode(w, r, nil, userID, code)
+		return
+	}
+	l.renderMailVerification(w, r, nil, userID, nil)
+}
+
+func (l *Login) handleMailVerificationCheck(w http.ResponseWriter, r *http.Request) {
+	data := new(mailVerificationFormData)
+	authReq, err := l.getAuthRequestAndParseData(r, data)
+	if err != nil {
+		l.renderError(w, r, authReq, err)
+		return
+	}
+	if !data.Resend {
+		l.checkMailCode(w, r, authReq, data.UserID, data.Code)
+		return
+	}
+	userOrg := "LOGIN"
+	if authReq != nil {
+		userOrg = authReq.UserOrgID
+	}
+	err = l.authRepo.ResendEmailVerificationMail(setContext(r.Context(), userOrg), data.UserID)
+	l.renderMailVerification(w, r, authReq, data.UserID, err)
+}
+
+func (l *Login) checkMailCode(w http.ResponseWriter, r *http.Request, authReq *model.AuthRequest, userID, code string) {
+	//TODO: Where do we get org id, if its not in login flow
```

Comment:
@fgerschwiler shall we remove the todo?

Reply:
yes we can, then we should put "LOGIN" in a const